syntax = "proto3";

package data;

// Request message containing CSV data as bytes.
message DataRequest {
    bytes csv_content = 1;
}


// Response message containing cleaned data.
message DataResponse {
    repeated double x_train = 1;
    repeated double x_test = 2;
    repeated double y_train = 3;
    repeated double y_test = 4;
    repeated string dates_train = 5;
    repeated string dates_test = 6;
}


// The DataService definition.
service DataService {
    // Sends CSV content and receives cleaned data.
    rpc CleanData (DataRequest) returns (DataResponse);
}
